src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.c

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "jni.h"
+#include "jdwpTransport.h"
+#include "fileSocketTransport.h"
+
+#ifdef _WIN32
+#include <winsock2.h>
+#else
+#include <arpa/inet.h>
+#endif
+
+#define MAX_FILE_SOCKET_PATH_LEN 4096
+#define MAX_DATA_SIZE 1000
+#define HANDSHAKE "JDWP-Handshake"
+
+static jboolean fake_open = JNI_FALSE;
+static jboolean initialized = JNI_FALSE;
+static JavaVM *jvm;
+static char path[MAX_FILE_SOCKET_PATH_LEN + 1];
+static jdwpTransportCallback *callback;
+static char last_error[2048];
+static struct jdwpTransportNativeInterface_ nif;
+static jdwpTransportEnv single_env = (jdwpTransportEnv) &nif;
+
+void log_error(char const* format, ...) {
+    char* tmp = (*callback->alloc)(sizeof(last_error));
+    va_list ap;
+
+    if (tmp != NULL) {
+        va_start(ap, format);
+        vsnprintf(tmp, sizeof(last_error) - 1, format, ap);
+        tmp[sizeof(last_error) - 1] = '\0';
+        va_end(ap);
+
+        printf("Error: %s\n", tmp);
+
+        memcpy(last_error, tmp, sizeof(last_error));
+        (*callback->free)(tmp);
+    } else {
+        printf("Could not get memory to print error.\n");
+    }
+}
+
+static jdwpTransportError JNICALL fileSocketTransport_GetCapabilities(jdwpTransportEnv* env, JDWPTransportCapabilities *capabilities_ptr) {
+    JDWPTransportCapabilities result;
+
+    memset(&result, 0, sizeof(result));
+    result.can_timeout_attach = JNI_FALSE;
+    result.can_timeout_accept = JNI_FALSE;
```

Comment:
You can remove the setting of explicit fields to JNI_FALSE. The memset above already did it for you. I think this comes due to copy&Paste from socketTransport.c, but there these capabilities are set to JNI_TRUE

Reply:
I would like to keep it as it is, since it makes clear that we don't support timeouts. With just the memset it is hard to see this.